socket_connection.py

```diff
@@ -1,75 +1,152 @@
+        self.username = username
+        self.password = password
+        self.session_id = None
+        self.csrftoken = None
+        self.csrfmiddlewaretoken = None
+        self._get_cookies()
+        self._login()
+
+    def _get_cookies(self) -> None:
+        response = self.make_get_request(
+            path='/accounts/login/',
+            headers={
+                'Host': self.host,
+                'Connection': 'close',
+            }
+        )
+        sessionid_regex: str = r'sessionid=(\w+)'
+        sessionid_pattern = re.compile(sessionid_regex)
+
+        csrftoken_regex: str = r'csrftoken=(\w+)'
+        csrftoken_pattern = re.compile(csrftoken_regex)
+
+        csrfmwtoken_regex: str = r'name="csrfmiddlewaretoken" value="(.+)"'
+        csrfmwtoken_pattern = re.compile(csrfmwtoken_regex)
+
+        try:
+            sessionid: str = re.findall(sessionid_pattern, response)[0]
+            self.session_id = sessionid
+
+        except Exception as e:
+            print('An error occurred while parsing sessionid', e)
+
+        try:
+            csrftoken: str = re.findall(csrftoken_pattern, response)[0]
+            self.csrftoken = csrftoken
+
+        except Exception as e:
+            print('An error occurred while parsing csrftoken', e)
+
+        try:
+            csrfmwtoken: str = re.findall(csrfmwtoken_pattern, response)[0]
+            self.csrfmiddlewaretoken = csrfmwtoken
+
+        except Exception as e:
+            print('An error occurred while parsing csrfmwtoken', e)
+
+    def _parse_response_code(self, line: str) -> int:
+        response_code_regex: str = r'HTTP/1.1 ([0-9]{3})'
+        response_code_pattern = re.compile(response_code_regex)
+
+        try:
+            response_code: int = re.findall(response_code_pattern, line)[0]
+            return response_code
+
+        except Exception as e:
+            print('An error occurred while parsing response code', e)
+
+        return 0
+
+    def _login(self):
+        sock = self.create_socket_connection()
+        request_headline = ' '.join(['POST', '/accounts/login/', self.http_version])
+
+        body = f'username={self.username}&password={self.password}&csrfmiddlewaretoken={self.csrfmiddlewaretoken}&next=' + CLRF
+        headers = {
+            'Host': self.host,
+            'Cookie': f'sessionid={self.session_id}; csrftoken={self.csrftoken}',
+            'Content-Length': len(body.encode()),
+            'Content-Type': 'application/x-www-form-urlencoded',
+            'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8',
+            'Accept-Encoding': 'gzip, deflate, compress',
+            'Referer': f'https://{self.host}/accounts/login/',
+            'Connection': 'close'
+        }
+
+        payload = request_headline + CLRF + CLRF.join(f'{k}: {v}' for k, v in headers.items()) + (CLRF * 2) + body
+        sock.sendall(payload.encode())
+        self.socket_recv_all(sock)
```

Comment:
`socket_recv_all` returns the response, but `_login()` is just an internal operation and the return values can just be thrown away since we don't need to use it anywhere.